reframe/REFRAME_HTTP_TRANSPORT.md

```diff
@@ -16,36 +16,74 @@ The Reframe over HTTP protocol is defining the transport and message
 
-# Organization of this document
-
-- [HTTP Transport Design](#http-transport-design)
-  - [HTTP Caching Considerations](#http-caching-considerations)
-    - [POST vs GET](#post-vs-get)
-    - [Avoiding sending the same response messages twice](#avoiding-sending-the-same-response-messages-twice)
-    - [Client controls for time-based caching](#client-controls-for-time-based-caching)
-    - [Rate-limiting non-cachable POST requests](#rate-limiting-non-cachable-post-requests)
+## Organization of this document
+
+- [HTTP Endpoint](#http-endpoint)
+  - [Content type](#content-type)
+  - [HTTP methods](#http-methods)
+  - [Other notes](#other-notes)
+- [HTTP Caching Considerations](#http-caching-considerations)
+  - [POST vs GET](#post-vs-get)
+  - [Avoiding sending the same response messages twice](#avoiding-sending-the-same-response-messages-twice)
+  - [Client controls for time-based caching](#client-controls-for-time-based-caching)
+  - [Rate-limiting non-cachable POST requests](#rate-limiting-non-cachable-post-requests)
 - [Implementations](#implementations)
 
-# HTTP Transport Design
+## HTTP Endpoint
 
-All messages sent in HTTP body MUST be encoded as DAG-JSON and use explicit content type `application/vnd.ipfs.rpc+dag-json; version=1`
+```
+https://rpc-service.example.net/reframe
+```
+
+URL of a Reframe endpoint must end with `/reframe` path.
+
+### Content type
+
+Requests SHOULD be sent with explicit `Accept` and `Content-Type` HTTP headers specifying the body format.
+
+All messages sent in HTTP body MUST be encoded as either:
+
+- [DAG-CBOR](https://ipld.io/specs/codecs/dag-cbor/spec/), and use explicit content type `application/vnd.ipfs.rpc+dag-cbor; version=1`
+  - **This is a CBOR (binary) format for use in production.**
+  - CBOR request MUST include HTTP header: `Accept: application/vnd.ipfs.rpc+dag-cbor; version=1`
+  - CBOR request AND response MUST include header: `Content-Type: application/vnd.ipfs.rpc+dag-cbor; version=1`
+- [DAG-JSON](https://ipld.io/specs/codecs/dag-json/spec/), and use explicit content type `application/vnd.ipfs.rpc+dag-json; version=1`
+  - **This is a human-readable plain text format for use in testing and debugging.**
+  - JSON request MUST include header: `Accept: application/vnd.ipfs.rpc+dag-json; version=1`
+  - JSON request AND response MUST include header: `Content-Type: application/vnd.ipfs.rpc+dag-json; version=1`
+
+Implementations SHOULD error when an explicit content type is missing, but MAY decide to implement some defaults instead.
+The rules around implicit content type are as follows:
+
+- Requests without a matching `Content-Type` header MAY be interpreted as DAG-JSON.
+- Requests without a matching `Accept` header MAY produce a DAG-JSON response.
+- Responses without a matching `Content-Type` header MAY be interpreted as DAG-JSON.
+
+### HTTP methods
 
 Requests MUST be sent as either:
 
+- `GET /reframe/{mbase64url-dag-cbor}`

```

Comment:
Talked with @guseggert and it would be beneficial to caching / scaling if we had request type in the path – allows for custom routing/caching per method without having to deserialize JSON/CBOR:

```suggestion
- `GET /reframe/{FindProvidersRequest}/{mbase64url-dag-cbor}`

```


TBD is this is possible to implement in https://github.com/ipld/edelweiss, dropping this comment so we don't forget (if we decide to keep Reframe).

Reply:
Added in https://github.com/ipfs/specs/pull/327/commits/e1448c2eea5be30c62f86c0e81b94e59fe6cae11